src/renderer/containers/Onboard/CreatePassword/CreatePassword.tsx

```diff
@@ -0,0 +1,87 @@
+import * as React from "react";
+import {Component, ReactElement} from "react";
+import {IInputFormProps} from "../../../components/Input/InputForm";
+import {ButtonPrimary} from "../../../components/Button/ButtonStandard";
+import {History} from "history";
+import {MultipleInputVertical} from "../../../components/MultipleInputVertical/MultipleInputVertical";
+import passwordComplexity from "../../../services/validation/password-complexity";
+import {partition} from "../../../services/utils/utils";
+
+interface IState {
+    input: string;
+    confirmInput: string;
+    valid?: boolean;
+    validationMessage: string;
+    confirmed?: boolean;
+}
+
+const SEPARATOR = "and";
+
+export default class CreatePassword extends Component<{ history: History }, {}> {
+    public state: IState = {
+        input: "",
+        confirmInput: "",
+        valid: undefined,
+        validationMessage: "",
+        confirmed: undefined
+    };
+
+    public handleChange = (e: React.FormEvent<HTMLInputElement>): void => {
+        // convert empty string to arbitrary string for validation purposes
+        const input = e.currentTarget.value === "" ? "a" : e.currentTarget.value;
+        this.setState({input: input});
+        // validate input
+        const validation = passwordComplexity().required().validate(input);
+        const isValid = (!validation.error);
+        this.setState({valid: isValid});
+        // extract and set validation message
+        if (!isValid) {
+            const [baseError, complexityErrors] =
+                partition(validation.error.details,arg => arg.type.includes("length"));
+            this.setState({validationMessage:
+                    "Password must " +
+                    `${baseError.length > 0 ? `${baseError[0].message} ${SEPARATOR} ` : ""}`+
+                    `contain at least ${complexityErrors.map(v => v.message).join(` ${SEPARATOR} `)}`
+            });
+        }
+        if (this.state.confirmed !== undefined) {
+            this.setState({confirmed: (this.state.confirmInput === input)});
+        }
+    };
+
+    public compareInput = (e: React.FormEvent<HTMLInputElement>): void => {
+        this.setState({confirmInput: e.currentTarget.value});
+        this.setState({confirmed: (this.state.input === e.currentTarget.value)});
+    };
+
+    public render(): ReactElement {
+        const inputs: Array<IInputFormProps> = [
+            {
+                inputId:"inputPassword",
+                onChange: this.handleChange,
+                placeholder: "Enter password",
+                valid: this.state.valid,
+                errorMessage: this.state.validationMessage
+            },
+            {
+                inputId:"confirmPassword",
+                onChange: this.compareInput,
```

Comment:
why not use joi validation for confirm?

Reply:
I could use Joi validation as ``` Joi.string().valid(this.state.input).validate(e.currentTarget.value) ```
but it wouldn't change the need for saving this confirmation password string in the state (_there are few input flows that would brake otherwise_). 
That is why I decided to go with a simple comparison because I think it is overkill to use Joi to see if two values are the same? 
I can show you tomorrow which flows are problematic.

Reply:
but if you have state like this:
```
{
   form: {
    password: string,
    confirm: string
}
}
```
you can pass entire form to joi and use joi.ref for comparing